### PR TITLE
Rename unbacked SymInt prefix to u

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #117862
* __->__ #117859

Currently, it conflicts with Inductor's naming convention for index
variables

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by lezcano — APPROVED** [review]

File: third_party/kineto
Comment (lezcano) [inline third_party/kineto:1]: oops

**Review by jansel — DISMISSED** [review]
Change seems fine, but tests are failing.

**Comment by ezyang** [comment]
I nailed the export failure, someone was regexing the string name lol

File: torch/_inductor/codegen/common.py [hunk 1]
```diff
@@ -1316,6 +1316,7 @@ def rename_indexing(self, index) -> sympy.Expr:
             x: self.args.size(x)
             for x in sorted_symbols
             if x.name.startswith("s")
+            or x.name.startswith("u")
             or x.name.startswith("ps")
```
Comment (Skylion007) [inline torch/_inductor/codegen/common.py:1320]:
```suggestion
            if x.name.startswith(("s", "u", "ps"))
```

File: torch/fx/experimental/symbolic_shapes.py [hunk 2]
```diff
@@ -274,7 +274,7 @@ def has_free_symbols(val: Union[SymInt, torch.Tensor]) -> bool:
 # Like free_symbols, but filtered to only report unbacked symbols
 def free_unbacked_symbols(x):
     # NB: keep synced with is_unbacked_symint
-    return {s for s in free_symbols(x) if s.name.startswith(("i", "f"))}
+    return {s for s in free_symbols(x) if s.name.startswith(("u", "f"))}
```
Comment (avikchaudhuri) [inline torch/fx/experimental/symbolic_shapes.py:277]: Should there be / is there some documentation in this file on what the various possible starting chars in a symbol name mean? (I didn't know there were so many.)

**Review by avikchaudhuri — APPROVED** [review]

**Review by jansel — APPROVED** [review]

File: torch/fx/experimental/symbolic_shapes.py (hunk 2 above)
Comment (ezyang) [reply torch/fx/experimental/symbolic_shapes.py:277]: I would write it but even I don't know what they all are 😭

**Pull request closed without merging**